app/views/layouts/application.html.erb

```diff
@@ -2,13 +2,69 @@
 <html>
 <head>
   <title>Betsy</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+
+
   <%= csrf_meta_tags %>
 </head>
 <body>
 
+  <%= image_tag("hipsterly_logo_plain.png", class: "img-responsive", style: "height: 25rem; width: 50rem; margin: auto;") %>
+
+  <p><%= "Hi #{username}" %></p>
+  <nav class="navbar navbar-default">
+    <ul class="nav navbar-nav">
+      <li><%= link_to "Home", root_path %></li>
+      <li><%= link_to "My Account", user_orders_path(session[:user_id])  if current_user %></li>
+      <li><%= link_to "Log In", new_session_path unless current_user %></li>
+      <li><%= link_to "Log Out", logout_path, method: :delete if current_user%></li>
+      <li>
+        <%= link_to "Sign Up", new_user_path unless current_user %>
+      </li>
+      <li><%= link_to "Cart", cart_path %></li>
+    </ul>
+  </nav>
+
+<aside class="col-xs-3">
+
+  <h3 class=side-text-white><%= link_to "View all products", products_path%></h3>
+  <h2 class="side-text">Browse products by</h2>
+  <h3 class="side-text">Category:</h3>
+  <section>
+    <% @categories = Product.select('DISTINCT category').map(&:category)%>
+    <% @categories.each do |category| %>
+    <% if category %>
+      <li class=side-text-white><%= link_to category.upcase, by_category_products_path(category) %></li>
+      <% end %>
+    <% end %>
+  </section>
+
+  <h4>Sellers:</h4>
+  <section>
+    <% @sellers = User.all.select { |user| user.products.count > 0 } %>
```

Comment:
Nice job excluding all the users that don't have products listed yet. Again, make sure this logic happens in the controller, not in the view.